### PR TITLE
workflows/doctor: use ephemeral runners

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This workflow hasn't been set up to run on our ephemeral runners. Let's
fix that.

### CODE REVIEW

**Comment by Bo98** [comment]
I've enabled this repo in the GitHub App settings, so should work now.

File: .github/workflows/doctor.yml [hunk 1]
```diff
@@ -18,9 +18,15 @@ jobs:
...
     strategy:
       matrix:
-        version: ["12-arm64", "12", "11-arm64", "11", "10.15"]
+        include:
+          - runner: "13-arm64-${{github.run_id}}-${{github.run_attempt}}"
+          - runner: "12-arm64"
+          - runner: "12-${{github.run_id}}-${{github.run_attempt}}"
+          - runner: "11-arm64"
+          - runner: "11-${{github.run_id}}-${{github.run_attempt}}"
+          - runner: "10.15-${{github.run_id}}-${{github.run_attempt}}"
```
Comment (Bo98) [inline .github/workflows/doctor.yml:27]:
```suggestion
        runner:
          - "13-arm64-${{github.run_id}}-${{github.run_attempt}}"
          - "12-arm64"
          - "12-${{github.run_id}}-${{github.run_attempt}}"
          - "11-arm64"
          - "11-${{github.run_id}}-${{github.run_attempt}}"
          - "10.15-${{github.run_id}}-${{github.run_attempt}}"
```

**Comment by Bo98** [comment]
Some of it will be updating homebrew-core being slow (ephemeral will have an older copy while non-ephemeral will be near up-to-date from the last run) and needing to run `brew install-bundler-gems` (which we could cache).

Cleanup slowness on Monterey still needs further investigation.

**Comment by MikeMcQuaid** [comment]
> Some of it will be updating homebrew-core being slow (ephemeral will have an older copy while non-ephemeral will be near up-to-date from the last run)

Perhaps we could avoid doing this at all for Homebrew/brew usage?

> needing to run `brew install-bundler-gems` on an older install (which we could cache).

actions/cache feels like a good call. also: why do we need these gems for `brew doctor`?

> Cleanup slowness on Monterey still needs further investigation.

Is this pre-cleanup?

**Comment by Bo98** [comment]
> why do we need these gems for `brew doctor`?

We don't really but it's a fixed part of `--only-setup`.

> Is this pre-cleanup?

Yes. We could skip it, but I'm a bit concerned something that should be doing little to nothing is taking 3 minutes so this feels like something's wrong.

**Comment by MikeMcQuaid** [comment]
> We don't really but it's a fixed part of `--only-setup`.

Gotcha. Could definitely remove or move to another `--only` where these gems are actually needed.

> We could skip it, but I'm a bit concerned something that should be doing little to nothing is taking 3 minutes so this feels like something's wrong that's indicating something to fix in the base image.

Agreed 👍🏻.

IMO we should do the bare minimum to only cleanup the stuff `brew doctor` complains about that. Could experiment with faster ways of doing this e.g. renaming `/usr/local/include` to `/usr/local/include.old` instead of moving to `/tmp`.

**Comment by carlocab** [comment]
> > We don't really but it's a fixed part of `--only-setup`.
> 
> Gotcha. Could definitely remove or move to another `--only` where these gems are actually needed.

How much of the setup do we need? (Or the cleanup steps, for that matter?) I'm wondering if we can cut those all out and run `setup-homebrew` and then call `brew doctor`.

**Comment by Bo98** [comment]
> IMO we should do the bare minimum to only cleanup the stuff `brew doctor` complains about that. Could experiment with faster ways of doing this e.g. renaming `/usr/local/include` to `/usr/local/include.old` instead of moving to `/tmp`.

The self-hosted images should already be set up that the cleanup required is zero.

> How much of the setup do we need?

Should be zero I think.

**Comment by carlocab** [comment]
So, maybe something like this then?
```diff
diff --git a/.github/workflows/doctor.yml b/.github/workflows/doctor.yml
index 50409557e..0335cda25 100644
--- a/.github/workflows/doctor.yml
+++ b/.github/workflows/doctor.yml
@@ -32,8 +32,9 @@ jobs:
         uses: Homebrew/actions/setup-homebrew@master
 
       - run: brew test-bot --only-cleanup-before
+        if: ! contains(matrix.runner, github.run_id)
 
-      - run: brew test-bot --only-setup
+      - run: brew doctor
 
       - run: brew test-bot --only-cleanup-after
-        if: always()
+        if: always() && ! contains(matrix.runner, github.run_id)
```
Not sure if some of those should be wrapped in `${{ }}`. I assume not.

**Pull request merged**